device_test.go

```diff
@@ -452,3 +452,68 @@ func TestRescanPciBus(t *testing.T) {
 	err := rescanPciBus()
 	assert.Nil(err)
 }
+
+func TestFindBlkCCWDevPath(t *testing.T) {
+	bus := "0.0.0005"
+	assert := assert.New(t)
+	expectDev := "vdb"
+	dir := fmt.Sprintf("/tmp/sys/bus/ccw/devices/%s/virtio5/block/%s", bus, expectDev)
+	busPath := fmt.Sprintf("/tmp/sys/bus/ccw/devices/%s", bus)
+	err := os.MkdirAll(dir, mountPerm)
+	assert.Nil(err)
+
+	dev, err := findBlkCCWDevPath(busPath)
+	assert.Nil(err)
+
+	if dev != expectDev {
+		t.Errorf("Expected value %s got %s", expectDev, dev)
+	}
+
+	bus = "../dev"
+	busPath = fmt.Sprintf("/tmp/sys/bus/ccw/devices/%s", bus)
+	err = os.MkdirAll(dir, mountPerm)
+	assert.Nil(err)
+	_, err = findBlkCCWDevPath(busPath)
+	assert.NotNil(err, fmt.Sprintf("findBlkCCWDevPath() should have been failed with bus %s", bus))
+
+}
+
+func TestCheckCCWBusFormat(t *testing.T) {
+	assert := assert.New(t)
+
+	bus := ""
+	err := checkCCWBusFormat(bus)
+	assert.NotNil(err, fmt.Sprintf("checkCCWBusFormat() should have been failed with bus %s", bus))
+
+	bus = "fe.0.0000"
+	err = checkCCWBusFormat(bus)
+	assert.NotNil(err, fmt.Sprintf("checkCCWBusFormat() should have been failed with bus %s", bus))
```

Comment:
Not a blocker but this test would be more compact and wouldn't have the duplication if you made it into a table-based test:

- https://github.com/kata-containers/tests/blob/master/Unit-Test-Advice.md#table-driven-tests

Reply:
much more elegant now